File: .cpp/14.LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int length = strs[0].size();
        bool check;
        string str="";
        char c;
        for(int i =0;i<length;i++)
        {
            int len = strs.size();
            c = strs[0][i]; 
            check=true;
            for(int j=1;j<len;j++)
            {
                if(c!=strs[j][i])
                {
                    check = false;
                    break;
                }
            }
            if(check)
            {
                str+=strs[0][i];
            }
            else{
                break;
            }
        }
        return str;
    }
};

File: .py/14.LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
       if not strs:
            return ""
        
       length = len(strs[0])
       result = ""
        
       for i in range(length):
            c = strs[0][i]
            check = True

            for j in range(1, len(strs)):
                if i >= len(strs[j]) or c != strs[j][i]:
                    check = False
                    break

            if check:
                result += c
            else:
                break

       return result
        